### PR TITLE
Update beautify extension

### PR DESCRIPTION
**Pull request by adamjimenez**
This extension had stopped working correctly.
This update is a complete rewrite and works in a less destructive way by preserving empty rows and whitespace within a row.
Tested with PHP, HTML, CSS and JS but should work with other modes as well.

### CODE REVIEW

File: lib/ace/ext/beautify.js [hunk 1]
```diff
@@ -34,15 +34,178 @@ define(function(require, exports, module) {
 "use strict";
 var TokenIterator = require("ace/token_iterator").TokenIterator;
 
-var phpTransform = require("./beautify/php_rules").transform;
```
Comment (nightwing) [inline lib/ace/ext/beautify.js]: do we need to delete https://github.com/ajaxorg/ace/blob/master/lib/ace/ext/beautify/php_rules.js if it is not used anymore?

Comment (adamjimenez) [reply lib/ace/ext/beautify.js]: yes

**Comment by nightwing** [comment]
Could you add a test to this, nothing fancy, just call the function with several different inputs, to make sure this is not broken again when modes are changed

**Comment by adamjimenez** [comment]
I've added some tests.

**Comment by nightwing** [comment]
Awesome, merged.

**Pull request merged**